src/uri/serializer/uriserializer.rs

```diff
@@ -66,8 +66,10 @@ pub trait UriSerializer<T> {
             return Err(SerializationError::new("Input uris are empty"));
         }
 
-        let long_uri = UUri::try_from(long_uri)?;
-        let micro_uri = UUri::try_from(micro_uri.to_vec())?;
+        let long_uri =
+            UUri::try_from(long_uri).map_err(|e| SerializationError::new(e.to_string()))?;
+        let micro_uri = UUri::try_from(micro_uri.to_vec())
+            .map_err(|e| SerializationError::new(e.to_string()))?;
```

Comment:
IMHO we should use different names for the variables here in order to prevent confusion with the passed in parameters ...

Reply:
Done